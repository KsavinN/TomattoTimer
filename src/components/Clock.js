import React from "react";
import PropTypes from "prop-types";

export default function Clock({ className,  hours, minutes , seconds , miliseconds }) {
    const properTime = (time) => {
      if (0 > time || time > 60) {
        return (time > 60) ? '59' : '00';
      }
      time = '0' + time;
       
      return (time.length > 2) ? time.slice(1) : time;
    }
    
  const properMinutes = (minutes) => <span className="clock__minutes">{properTime(minutes)}</span>
  
  const properSeconds = (seconds) => <span className="clock__seconds">{properTime(seconds)}</span>

    const properHour = (time) => {
      if (0 > time || time > 24) {
        return (time > 24) ? '24' : '00';
      }
      time = '0' + time;
      return <span className={`clock__hours`}>{(time.length > 2) ? time.slice(1) : time}</span>
    }
    // const properMili = (time) => {
    //   if (0 > time || time > 999) {
    //     return (time > 999) ? '999' : '000';
    //   }
    //   time = '00' + time;
    //   return (time.length > 3) ? time.slice(time.length - 3) : time;
    // }
    
    const separator = () => <span className={`clock__separator`}>:</span>
    
  return (<h2 className={"clock "}>
    {properHour(hours)}{separator()}{properMinutes(minutes)}{separator()}{properSeconds(seconds)}
    </h2>)
}
Clock.defaultProps = {
  className: "", 
  hours: 0,
  miliseconds:0
}
  
Clock.propTypes = {
  className: PropTypes.string.isRequired,
  minutes: PropTypes.number.isRequired,
  seconds: PropTypes.number.isRequired,
  hours: PropTypes.number.isRequired,
  miliseconds: PropTypes.number.isRequired
}