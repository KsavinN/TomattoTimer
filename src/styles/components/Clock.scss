.clock {

    &__hours {
        color: black;
    }

    &__minutes {
        color: lighten($color: #000000, $amount: 15);
    }

    &__seconds {
        color: lighten($color: #000000, $amount: 30);
    }

    &__separator {
        color: grey;
    }
}