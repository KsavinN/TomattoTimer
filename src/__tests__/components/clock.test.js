import React from 'react';
import Clock from "../../components/Clock"; 
import renderer from "react-test-renderer";

var clockRenderer = null;
describe('<Clock />', () => {
    describe('should properly when fiven minutes and seconds', () => {
        beforeEach(() => {
            clockRenderer = renderer.create(<Clock minutes={10} seconds={20} />)
        });
        it('renders properly', () => {
            expect(clockRenderer.toJSON()).toMatchSnapshot();
        })
        it('renders an h2 element', () => {
            expect(clockRenderer.toJSON().type).toEqual('h2')
        });
        it('sets a Clock className', () => {
            expect(clockRenderer.toJSON().props).toMatchObject({"className": expect.stringMatching(/clock/)})
        });
        it('renders minutes properly', () => {
            expect(clockRenderer.toJSON().children[2].children).toEqual(expect.arrayContaining(['10']))
        });
        it('renders seconds properly', () => {
            expect(clockRenderer.toJSON().children[4].children).toEqual(expect.arrayContaining(['20']))
        });
    });
});