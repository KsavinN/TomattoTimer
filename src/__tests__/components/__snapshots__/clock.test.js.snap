// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> should properly when fiven minutes and seconds renders properly 1`] = `
<h2
  className="clock "
>
  <span
    className="clock__hours"
  >
    00
  </span>
  <span
    className="clock__separator"
  >
    :
  </span>
  <span
    className="clock__minutes"
  >
    10
  </span>
  <span
    className="clock__separator"
  >
    :
  </span>
  <span
    className="clock__seconds"
  >
    20
  </span>
</h2>
`;
